### PR TITLE
feat(you-llm): Integrate You.com conversational APIs

### PR DESCRIPTION
**Pull request by christeefy**
You.com is releasing two new conversational APIs — Smart and Research.

This PR integrates those APIs as an LM.

### CODE REVIEW

File: docs/api/language_model_clients/You.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# dspy.You
```
Comment (arnavsinghvi11) [inline docs/api/language_model_clients/You.md:1]: Can you add 

```---
sidebar_position: 12
---
```
to the top of this doc? seems like I can't edit directly

**Comment by arnavsinghvi11** [comment]
Thanks @christeefy ! left a comment for the docs before it's ready to merge

**Comment by christeefy** [comment]
Made the changes, @arnavsinghvi11. PTAL!

**Comment by arnavsinghvi11** [comment]
Thanks @christeefy !

**Pull request merged**